api/envoy/api/v2/endpoint/endpoint.proto

```diff
@@ -74,6 +80,8 @@ message LbEndpoint {
   // to subset the endpoints considered in cluster load balancing.
   core.Metadata metadata = 3;
 
+  repeated Label labels = 6;
```

Comment:
@htuch

Reply:
So @htuch / @lizan I have been giving this some thought. I think its better to have a named TLS context rather than a label selector because named TLS contexts refer to exactly one context to use. Label selectors could select multiple contexts and then we would have to define some first match wins semantics which gets clunky.

Reply:
I'd like to this be first match wins (similar to FCM), the context map is couples EDS with CDS too tightly, and the matcher semantics here is more flexible to match a subset has two labels etc.

Reply:
Yeah, let's just have a `string` field here pointing at a specific TLS context. This will be the clearest I reckon.

Reply:
I'm fine with that.

Reply:
done.

Reply:
Switching back into this review after some time out, it seems that we still have a label match scheme. If the consensus is heading towards this being necessary,  there are two things I'd ask:
1. Document a clear use case of why we have this in `endpoint_transport_socket_matchers` comments.
2. Use the existing `metadata`; I originally pushed back on this as I felt we could just specify a transport socket name here, but now that we are going to a "label" scheme, we already have a source of this information, the endpoint metadata. There should be well-known key name for this, `envoy.transport_socket`.

Reply:
Done. Added a use case in the end of this field comments. @lizan fyi.

I also renamed the field by removing "endpoint_", since as @rshriram points out, it's possible to add metadata for DNS based endpoint as well.